### PR TITLE
Change external submissions from using static form to React form

### PR DESCRIPTION
**Pull request by martha**
## Description

GH issue: https://github.com/greenriver/hmis-warehouse/pull/4713#discussion_r1765474000
This PR moves the External Forms review UI from a static form to a directly configured React form, since the static form doesn't seem to be making development any easier.

## Type of change
[//]: # 'remove options that are not relevant'
- [ ] Bug fix
- [ ] New feature (adds functionality)
- [x] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/modules/projects/ExternalSubmissionsViewModal.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
+import {
```
Comment (gigxz) [inline src/modules/projects/ExternalSubmissionsViewModal.tsx]: can we move this up into the "components" dir?

File: src/modules/projects/ExternalSubmissionsViewModal.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+              value={status === ExternalFormSubmissionStatus.Reviewed}
+              label='Mark as Reviewed'
+              onChange={(_e, checked) => {
+                setStatus(
+                  checked
+                    ? ExternalFormSubmissionStatus.Reviewed
+                    : ExternalFormSubmissionStatus.New
+                );
+              }}
+            />
```
Comment (gigxz) [inline src/modules/projects/ExternalSubmissionsViewModal.tsx]: Assuming that a future PR on the PIT branch will hide this checkbox for forms that have been reviewed.

If we go with my suggestion from https://github.com/greenriver/hmis-frontend/pull/936, we could replace it with something like `This form has been reviewed [and here is the Enrollment it generated, if applicable]`

**Review by gigxz — APPROVED** [review]
This feels much nicer, thanks!

File: src/modules/projects/ExternalSubmissionsViewModal.tsx (hunk 1 above)
Comment (martha) [reply src/modules/projects/ExternalSubmissionsViewModal.tsx]: Yes, sorry!

File: src/modules/projects/ExternalSubmissionsViewModal.tsx (hunk 2 above)
Comment (martha) [reply src/modules/projects/ExternalSubmissionsViewModal.tsx]: Yes -- good plan.

**Comment by martha** [comment]
There was no separate ticket for this work outside of https://github.com/open-path/Green-River/issues/6598, the bulk of which is on `pit-external-forms` branch, but just adding a note that I qa'd this portion which was deployed with release-134 and it looks good.
- Can navigate to an external form submission
- Can submit review with a note and new status/spam value

**Pull request merged**